vir/test.cpp

```diff
@@ -53,6 +63,82 @@ namespace
 }
 #endif // VIR_HAVE_SIMD_CVT
 
+#if VIR_HAVE_SIMD_PERMUTE and VIR_HAVE_SIMD_IOTA
+static_assert(
+  all_of(vir::simd_permute(make_simd(0, 1, 2, 3), vir::simd_permutations::duplicate_even)
+	   == make_simd(0, 0, 2, 2)));
+
+static_assert(
+  all_of(vir::simd_permute(make_simd(0, 1, 2, 3), vir::simd_permutations::duplicate_odd)
+	   == make_simd(1, 1, 3, 3)));
+
+static_assert(
+  all_of(vir::simd_permute(make_simd(0, 1, 2, 3), vir::simd_permutations::swap_neighbors<>)
+	   == make_simd(1, 0, 3, 2)));
+
+static_assert(
+  all_of(vir::simd_permute(make_simd(0, 1, 2, 3), vir::simd_permutations::swap_neighbors<2>)
+	   == make_simd(2, 3, 0, 1)));
```

Comment:
At first this was surprising to me. (I had expected something else. I can't say exactly what though.)

Anyway: Are there simd-types (maybe with T=char) that have 6 elements, then maybe include an example with `vir::simd_permutations::swap_neighbors<3>` as well?

Reply:
good point swapping 2 on 4 elements is still ambiguous. will do